src/hooks/sidebar/useSidebarType.ts

```diff
@@ -1,63 +1,81 @@
-import { useState } from 'react';
-import { useDispatch } from 'react-redux';
+import { useDispatch, useSelector } from 'react-redux';
 import {
-  setFeature,
-  setSubFeature,
-  setElement,
-  setSubElement,
+  setSidebarProperties,
+  initSidebarProperties,
 } from '../../store/sidebar/action';
+import { RootState } from '../../store/index';
 import {
   FeatureNameType,
   ElementNameType,
   SubElementNameType,
+  PayloadPropsType,
+  SidebarProperties,
 } from '../../store/common/type';
 
 export interface SidebarHookType {
-  sidebarTypeClickHandler: (name: string) => void;
+  sidebarTypeClickHandler: (name: FeatureNameType | ElementNameType) => void;
   sidebarSubTypeClickHandler: (name: string) => void;
-  sidebarTypeName: string;
-  sidebarSubTypeName: string;
+  feature: FeatureNameType | null;
+  subFeature: string | null;
+  element: ElementNameType | null;
+  subElement: SubElementNameType | null;
 }
 
 function useSidebarType(): SidebarHookType {
-  const [sidebarTypeName, setSidebarTypeName] = useState<string>('');
-  const [sidebarSubTypeName, setSidebarSubTypeName] = useState<string>('');
   const dispatch = useDispatch();
+  const sidebarStates = useSelector<RootState>(
+    (state) => state.sidebar
+  ) as PayloadPropsType;
+  const { feature, subFeature, element, subElement } = sidebarStates;
 
-  const sidebarTypeClickHandler = (name: string) => {
-    if (name !== sidebarTypeName) {
-      if (
-        name === ElementNameType.section ||
-        name === ElementNameType.labelText ||
-        name === ElementNameType.labelIcon
-      ) {
-        dispatch(setElement(name as ElementNameType));
-      } else {
-        dispatch(setFeature(name as FeatureNameType));
-      }
-      setSidebarTypeName(name);
+  const sidebarTypeClickHandler = (name: FeatureNameType | ElementNameType) => {
+    if ([feature, element].includes(name)) return;
+    if (Object.keys(ElementNameType).includes(name)) {
```

Comment:
여기서 어제 제가 실수 했던 게  [in 연산자](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Operators/in) 인데,

```suggestion
    if (name in ElementNameType) {..}
```

이렇게 바꿨으면 됐을 것 같아요 ㅋㅋ

Reply:
원래는 어떻게 하셨었죠???

Reply:
원래는 `name in Object.keys(ElementNameType)` 으로 했어서 이렇게하면 배열의 key인 0부터 object length까지 범위에서 name을 찾게 돼요 ㅋㅋㅋㅋ